src/main/java/io/confluent/idesidecar/restapi/credentials/ScramCredentials.java

```diff
@@ -0,0 +1,111 @@
+package io.confluent.idesidecar.restapi.credentials;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.confluent.idesidecar.restapi.exceptions.Failure;
+import io.confluent.idesidecar.restapi.exceptions.Failure.Error;
+import io.soabase.recordbuilder.core.RecordBuilder;
+import jakarta.validation.constraints.NotNull;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+@Schema(description = "Scram authentication credentials")
+@RecordBuilder
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonSubTypes({
+    @JsonSubTypes.Type(value = ScramCredentials.class, name = "SCRAM")
+})
+public record ScramCredentials(
+    @Schema(description = "Hash algorithm")
+    @JsonProperty(value = "hash_algorithm")
+    @NotNull
+    HashAlgorithm hashAlgorithm,
+    @NotNull
+    String username,
+    @NotNull
+    Password password
+) implements Credentials {
+
+  @Override
+  public Type type() {
+    return Type.SCRAM;
+  }
+
+  private static final String SCRAM_LOGIN_MODULE_CLASS =
+      "org.apache.kafka.common.security.scram.ScramLoginModule";
+
+  @Override
+  public Optional<Map<String, String>> kafkaClientProperties(
+      KafkaConnectionOptions options
+  ) {
+    var jaasConfig = "%s required username=\"%s\" password=\"%s\"".formatted(
+        SCRAM_LOGIN_MODULE_CLASS,
+        username,
+        password.asString(options.redact())
+    );
+
+    // Terminate the JAAS configuration with a semicolon
+    jaasConfig += ";";
```

Comment:
I think we don't need to terminate the config in a separate line

```suggestion
    var jaasConfig = "%s required username=\"%s\" password=\"%s\";".formatted(
        SCRAM_LOGIN_MODULE_CLASS,
        username,
        password.asString(options.redact())
    );
```